api/handlers.go

```diff
@@ -678,6 +653,49 @@ func postCommit(c *context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// POST /build
+func postBuild(c *context, w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+		httpError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	buildImage := &dockerclient.BuildImage{}
```

Comment:
This could be prettier if inlined:

```
buildImage := &dockerclient.BuildImage{
	DockerfileName: r.Form.Get("dockerfile"),
	RepoName: r.Form.Get("t"),
...
}